lib/liquid/strainer_template.rb

```diff
@@ -3,54 +3,41 @@
 require 'set'
 
 module Liquid
-  # Strainer is the parent class for the filters system.
+  # StrainerTemplate is the computed class for the filters system.
   # New filters are mixed into the strainer class which is then instantiated for each liquid template render run.
   #
-  # The Strainer only allows method calls defined in filters given to it via Strainer.global_filter,
+  # The Strainer only allows method calls defined in filters given to it via StrainerFactory.add_global_filter,
   # Context#add_filters or Template.register_filter
-  class Strainer #:nodoc:
-    @@global_strainer = Class.new(Strainer) do
-      @filter_methods = Set.new
-    end
-    @@strainer_class_cache = Hash.new do |hash, filters|
-      hash[filters] = Class.new(@@global_strainer) do
-        @filter_methods = @@global_strainer.filter_methods.dup
-        filters.each { |f| add_filter(f) }
-      end
-    end
-
+  class StrainerTemplate
     def initialize(context)
       @context = context
     end
 
     class << self
-      attr_reader :filter_methods
-    end
+      def add_filter(filter)
+        raise ArgumentError, "Expected module but got: #{filter.class}" unless filter.is_a?(Module)
```

Comment:
We should probably add a test for this because

```
irb(main):007:0> Class.new.is_a?(Module)
=> true
```

This should probably be `filter.instance_of?(Module)` because 

```
irb(main):011:0> Class.ancestors
=> [Class, Module, Object, Kernel, BasicObject]
```

Reply:
I've simply removed it since we now call `include?` which will raise a `TypeError`.

This dated from before the code had this shape from the initial git import: https://github.com/Shopify/liquid/blame/1d647361e1a39ee09f1bd2d62701925c0128c578/lib/liquid/strainer.rb#L27